Stripe/STPAPIClient.m

```diff
@@ -80,40 +87,43 @@ - (void)createTokenWithData:(NSData *)data completion:(STPCompletionBlock)comple
     
-    STPAPIConnection *connection = [[STPAPIConnection alloc] initWithRequest:request];
-    
-    [connection runOnOperationQueue:self.operationQueue
-                         completion:^(NSURLResponse *response, NSData *body, NSError *requestError) {
-                             if (requestError) {
-                                 // If this is an error that Stripe returned, let's handle it as a StripeDomain error
-                                 if (body) {
-                                     NSDictionary *jsonDictionary = [NSJSONSerialization JSONObjectWithData:body options:0 error:NULL];
-                                     if ([jsonDictionary valueForKey:@"error"] != nil) {
-                                         completion(nil, [self.class errorFromStripeResponse:jsonDictionary]);
-                                         return;
-                                     }
-                                 }
-                                 completion(nil, requestError);
-                                 return;
-                             } else {
-                                 NSDictionary *jsonDictionary = [NSJSONSerialization JSONObjectWithData:body options:0 error:NULL];
-                                 if (!jsonDictionary) {
-                                     NSDictionary *userInfo = @{
-                                                                NSLocalizedDescriptionKey: STPUnexpectedError,
-                                                                STPErrorMessageKey: @"The response from Stripe failed to get parsed into valid JSON."
-                                                                };
-                                     NSError *error = [[NSError alloc] initWithDomain:StripeDomain code:STPAPIError userInfo:userInfo];
-                                     completion(nil, error);
-                                 } else if ([(NSHTTPURLResponse *)response statusCode] == 200) {
-                                     completion([[STPToken alloc] initWithAttributeDictionary:jsonDictionary], nil);
-                                 } else {
-                                     completion(nil, [self.class errorFromStripeResponse:jsonDictionary]);
-                                 }
-                             }
-                         }];
+    [[self.urlSession dataTaskWithRequest:request completionHandler:^(NSData * _Nullable body, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        if (error) {
+            // If this is an error that Stripe returned, let's handle it as a StripeDomain error
+            if (body) {
+                NSDictionary *jsonDictionary = [NSJSONSerialization JSONObjectWithData:body options:0 error:NULL];
+                if ([jsonDictionary valueForKey:@"error"] != nil) {
+                    [self.operationQueue addOperationWithBlock:^{
```

Comment:
I agree it's gross. It's because the `operationQueue` can be set after the fact, so if you initialize an API client and then change its operation queue, it won't update the underlying url session (since that property can only be set during initialization on NSURLSession). Really we should probably just deprecate this property.